tests/testthat/test-annotations.R

```diff
@@ -0,0 +1,92 @@
+    threstype = "above threshold", threshold = "0.05", color = "blue",
+    size = 20, width = 1, offset = 0, font.family = "Arial",
+    font.weight = "normal", font.style = "normal"))
+
+a2 <- list(list(type = "Circle - filled", data = "p", threstype = "below threshold",
+    threshold = "0.10", color = "#C0C0C0", size = 25, width = NULL,
+    offset = NULL, font.family = NULL, font.weight = NULL, font.style = NULL),
+    list(type = "Circle - filled", data = "p", threstype = "below threshold",
+        threshold = "0.05", color = "#80FFFF", size = 20, width = NULL,
+        offset = NULL, font.family = NULL, font.weight = NULL,
+        font.style = NULL), list(type = "Arrow - up", data = "p",
+        threstype = "below threshold", threshold = "0.05", color = "#CD343C",
+        size = 15, width = NULL, offset = NULL, font.family = NULL,
+        font.weight = NULL, font.style = NULL))
+a3 <- list(list(type = "Circle - filled", data = "p", threstype = "above threshold",
+    threshold = "-Inf", color = "red", size = 20, width = 1,
+    offset = 0, font.family = "Arial", font.weight = "normal",
+    font.style = "normal"), list(type = "Circle - thin outline", data = "p",
+    threstype = "above threshold", threshold = "0.05", color = "blue",
+    size = 20, width = 1, offset = 0, font.family = "Arial",
+    font.weight = "normal", font.style = "normal"))
+
+a4 <- list(list(type = "Text - after data labels", data = "p", threstype = "above threshold", threshold = "-Inf", color = "red", size = 8, width = 1, font.family = "Arial", font.weight = "normal", font.style = "normal", format = ".3f", prefix = " +/-"))
+
+test_that("Annotations",
+{
+    expect_error(Column(data.with.stats, data.label.show = TRUE, annotation.list=list(list(data="p", type = "Arrow - up", threstype = "above threshold", threshold = 0.05, color = "#FF0000", size = 15))), NA)
+
+    expect_error(Column(vec.with.stats, data.label.show = TRUE, annotation.list=list(list(data="p", type = "Up arrow", threstype = "above threshold", threshold = 0.05, color = "#0000FF"))), NA)
+
+    expect_error(Column(vec.with.stats, data.label.show = TRUE, annotation.list = list(list(data = "p",
+    type = "Circle - filled", size = 30, color = "red", threstype = "above threshold", threshold = 0.05), list(data = "p", type = "Circle - thin outline", size = 30, color = "blue"))), NA)
+
+    expect_error(Column(vec.with.stats, data.label.show = TRUE, annotation.list = list(list(data = "p",
+    type = "Filled circle", size = 35, color = "blue"), list(data = "p", type = "Filled circle", size = 30, color = "red"))), NA)
+
+    expect_error(Column(vec.with.stats, data.label.show = TRUE, annotation.list = list(list(data = "p",
+    type = "Circle - filled", size = 40, color = "red"), list(data = "p", type = "Circle - filled", size = 30, color = "orange"), list(data = "p", type = "Circle - filled", size = 20, color = "yellow"))), NA)
```

Comment:
I don't see annotations from this either?

Reply:
I've fixed the tests now and updated the code creating the annotation to give a warning and skip the annotation if the type is unrecognized. I'll add more tests to flipChartTests once this is merged into master to make sure the annotation output looks correct.

I've also made added functionality to move the circles horizontally. You can see this using the dev server here: https://app.displayr.com/Dashboard?project_id=-722594#page=b3460318-88ce-41c5-9e98-20bb331d5276